light-client/src/operations/commit_validator.rs

```diff
@@ -39,7 +46,39 @@ impl CommitValidator for ProdCommitValidator {
             )));
         }
 
-        signed_header.validate(&validator_set)?;
+        Ok(())
+    }
+
+    // This check is only necessary if we do full verification (2/3)
+    //
+    // See https://github.com/informalsystems/tendermint-rs/issues/281
+    //
+    // It returns `ImplementationSpecific` error if it detects a signer
+    // that is not present in the validator set
+    fn validate_full(
```

Comment:
it makes sense to decouple these and have a separate `validate_full` method instead 👍